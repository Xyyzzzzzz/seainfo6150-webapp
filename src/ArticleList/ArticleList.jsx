import React from "react";
import ArticleListItem from '../ArticleListItem/ArticleListItem'
import { isEmpty } from "lodash";

const ArticleList = ({ articleList }) => {
    const displayContent =
        isEmpty(articleList) ?
            <div>You have no data!</div> :
            <ul>
                {articleList.map((item, index) =>
                    <li key={index}>{ArticleListItem(item)}</li>
                )}
            </ul>

    return displayContent;
};

export default ArticleList;
