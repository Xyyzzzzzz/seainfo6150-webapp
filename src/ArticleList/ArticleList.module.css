.articleList{
    font-family:Arial, Helvetica, sans-serif;
    list-style: none;
    background-color:rgb(130, 140, 140);
    border: 1px dashed rgb(85, 85, 85);
    border-radius: 10px;
    box-shadow: 0px 0px 5px 5px rgba(255, 0, 0, 0.5);
    padding: 10px; 
    display: grid;
    grid-template-columns: repeat(3, 32%);
    justify-content: space-around;
}

@media (min-width: 400px) {
  .articleList {
      font-size: small;
      grid-template-columns: repeat(1, 90%);
    }
  }
  
  @media (min-width: 600px) {
    .articleList {
      font-size: medium;
      grid-template-columns: repeat(2, 48%);
    }
  }

  @media (min-width: 1000px) {
    .articleList {
      font-size: large;
      grid-template-columns: repeat(3, 32%);
    }
  }
  
  @media (min-width: 1500px) {
    .articleList{
      font-size: x-large;
      grid-template-columns: repeat(4, 23%);
      
    }
  }