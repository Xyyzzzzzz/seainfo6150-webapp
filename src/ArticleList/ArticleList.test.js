import React from "react";
import ArticleList from "./ArticleList";

const articleList = [{
    title: "test article title",
    shortText: "test article shortText",
    timeStamp: "2018-11-22T15:12:24.000Z",
    displayDate: "November 22nd 2018, 7:12 am"
}, {
    title: "test article title2",
    shortText: "test article shortText2",
    timeStamp: "2018-11-22T15:12:24.000Z",
    displayDate: "November 22nd 2018, 7:12 am"
}]

describe("ArticleList tests", () => {
    it("renders correctly", () => {
        const { container } = render(<ArticleList articleList={articleList} />);
        expect(container).toMatchSnapshot();
    });
});
