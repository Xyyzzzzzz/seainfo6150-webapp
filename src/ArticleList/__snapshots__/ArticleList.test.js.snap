// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList tests renders correctly 1`] = `
<div>
  <ul
    class="articleList"
  >
    <li>
      <div
        class="articleItem"
      >
        <h1
          class="articleTitle"
        >
          test article title
        </h1>
        <button
          class="toggleButton"
        >
          Show more
        </button>
      </div>
    </li>
    <li>
      <div
        class="articleItem"
      >
        <h1
          class="articleTitle"
        >
          test article title2
        </h1>
        <button
          class="toggleButton"
        >
          Show more
        </button>
      </div>
    </li>
  </ul>
</div>
`;
